src/messages/MMgrConfigure.h

```diff
@@ -35,19 +36,25 @@ class MMgrConfigure : public MessageInstance<MMgrConfigure> {
   // Default 0 means if unspecified will include all stats
   uint32_t stats_threshold = 0;
 
+  std::list<OSDPerfMetricQuery> osd_perf_metric_queries;
```

Comment:
So if we want `MMgrReport` to return query_id (instead of query itself) then we need to provide these ids somehow here. E.g. either change `std::list<OSDPerfMetricQuery>` to `std::map<OSDPerfMetricQueryID, OSDPerfMetricQuery>` or embed `OSDPerfMetricQueryID` into `OSDPerfMetricQuery`.

Reply:
Sure, so I embedded the id in the `OSDPerfMetricQuery` as you can see in commit  https://github.com/ceph/ceph/pull/24180/commits/ba69098f1ed50c62c1ab8946d55913d6c2f984e6